src/status_im2/common/floating_button_page/view.cljs

```diff
@@ -0,0 +1,113 @@
+
+(defn show-button-background
+  [keyboard-height keyboard-shown content-scroll-y]
+  (let [button-container-height 64
+        keyboard-view-height    (+ keyboard-height button-container-height)]
+    (when keyboard-shown
+      (cond
+        platform/android?
+        (< (- @scroll-view-height button-container-height) @content-container-height)
+
+        platform/ios?
+        (< (- @scroll-view-height keyboard-view-height) (- @content-container-height content-scroll-y))
+
+        :else
+        false))))
+
+(defn button-container
+  [{:keys [show-keyboard? keyboard-shown show-background? keyboard-height]} children]
+  (let [insets (safe-area/get-insets)
+        height (reagent/atom 0)]
+    (reset! height (if show-keyboard? (if keyboard-shown keyboard-height 0) 0))
+    [rn/view {:style {:margin-top :auto}}
+     (cond
+       (and (> @height 0) show-background?)
+       [blur/view
+        (when keyboard-shown
+          {:blur-amount      34
+           :blur-type        :transparent
+           :overlay-color    :transparent
+           :background-color (if platform/android? colors/neutral-100 colors/neutral-80-opa-1-blur)
+           :style            (style/blur-button-container insets)})
+        children]
+
+       (and (> @height 0) (not show-background?))
+       [rn/view {:style (style/view-button-container true insets)}
+        children]
+
+       (not show-keyboard?)
+       [rn/view {:style (style/view-button-container false insets)}
+        children])]))
+
+(defn f-view
+  [{:keys [button-props button-label]} header children]
+  (reagent/with-let [show-keyboard?   (reagent/atom false)
+                     content-scroll-y (reagent/atom 0)
+                     show-listener    (oops/ocall rn/keyboard
+                                                  "addListener"
+                                                  (if platform/android?
+                                                    "keyboardDidShow"
+                                                    "keyboardWillShow")
+                                                  #(reset! show-keyboard? true))
+                     hide-listener    (oops/ocall rn/keyboard
+                                                  "addListener"
+                                                  (if platform/android?
+                                                    "keyboardDidHide"
+                                                    "keyboardWillHide")
+                                                  #(reset! show-keyboard? false))]
+
+    (let [{:keys [keyboard-shown
+                  keyboard-height]} (hooks/use-keyboard)
+          show-background?          (show-button-background keyboard-height
+                                                            keyboard-shown
+                                                            @content-scroll-y)]
+
+      [rn/view {:style style/page-container}
+       header
+       [rn/scroll-view
```

Comment:
Will it ever need to be a flatlist?

Reply:
not quite sure, I talked to design and they said this type of page is needed when
'pretty much any scrollable page that has buttons on bottom have a "safety" background that varies depending on theme'
I think the flat list can be within this but the scroll behaviour is right this way. 
wdyt?

Reply:
If so far a FlatList is not needed for such a screen let's keep it that way